src/jit/valuenum.cpp

```diff
@@ -1188,7 +1265,7 @@ ValueNum ValueNumStore::VNForMapSelectWork(ValueNumKind vnk, var_types typ, Valu
     VNDefFunc2Arg fstruct(VNF_MapSelect, arg0VN, arg1VN);
     if (GetVNFunc2Map()->Lookup(fstruct, &res))
     {
-        return res;
+        goto Return;
```

Comment:
Many programmers prefer early returns. In one of my previous code reviews @BruceForstall suggested using an early return to avoid an extra nesting level. The existing code also uses early returns everywhere. So, I think whatever convention we decide to adopt, it should be in the coding conventions.
Since this is not essential to the bug fix, I will remove the dumping changes so no new goto's will be introduced.